### PR TITLE
NuGet packaging updates for 2.0.0 NuGet packages

### PR DESCRIPTION
**Pull request by jasonmalinowski**
Fixes two issues. This will fix future builds going forward, and we'll do a manual rerun of the script for 2.0.0.

- The Elfie package was still a pre-release version, so we bump it. The new package is the exact same contents as 0.10.6-rc2 but just repacked.
- We need to bump our Microsoft.Net.Compilers package since we accidentally uploaded a 2.0.0 package a long time ago.

*Review;* @dotnet/roslyn-ide, @CyrusNajmabadi, @jinujoseph, @jaredpar (for verifying I'm doing this right)

### CODE REVIEW

**Comment by jasonmalinowski** [comment]
I've revised the PR. @Pilchie, @jinujoseph, @tannergooding and others can we get a re-review of this? This can be merged in for future changes, once we get the Elfie package uploaded. In the meantime I've ran a manual rerun of the script (hacking it to explicitly look at the drop server) so we can repack for 2.0.0.

File: src/NuGet/BuildNuGets.csx [hunk 1]
```diff
@@ -212,8 +213,27 @@ int PackFiles(string[] nuspecFiles, string licenseUrl)
 
         if (!IsCoreBuild)
         {
+            string packageArgs = commonArgs;
+
+            // HACK: since Microsoft.Net.Compilers 2.0.0 was uploaded by accident and later deleted, we must bump the minor.
+            // We will do this to both the regular Microsoft.Net.Compilers package and also the netcore package to keep them
+            // in sync.
+            if (BuildVersion.StartsWith("2.0.") && Path.GetFileName(file).StartsWith("Microsoft.Net.Compilers."))
```
Comment (Pilchie) [inline src/NuGet/BuildNuGets.csx]: Case-sensitivity?

File: src/NuGet/BuildNuGets.csx [hunk 2]
```diff
@@ -212,8 +213,27 @@ int PackFiles(string[] nuspecFiles, string licenseUrl)
...
+            {
+                string[] buildVersionParts = BuildVersion.Split('-');
+                string[] buildVersionBaseParts = buildVersionParts[0].Split('.');
+                
+                buildVersionBaseParts[buildVersionBaseParts.Length - 1] =
+                    (int.Parse(buildVersionBaseParts[buildVersionBaseParts.Length - 1] + 1)).ToString();
```
Comment (Pilchie) [inline src/NuGet/BuildNuGets.csx]: Parentheses in the wrong spot  - aren't you concatenating the `1` here?

File: build/Targets/Dependencies.props [hunk 3]
```diff
@@ -15,7 +15,7 @@
     <MicrosoftBuildTasksCoreVersion>15.1.0-preview-000458-02</MicrosoftBuildTasksCoreVersion>
```
Comment (tannergooding) [inline build/Targets/Dependencies.props]: Not part of this PR strictly speaking, but we do have an RTW package available for this dependency now.

File: build/Targets/Dependencies.props [hunk 4]
```diff
@@ -101,4 +101,4 @@
     <xunitrunnerconsoleVersion>2.2.0-beta4-build3444</xunitrunnerconsoleVersion>
```
Comment (tannergooding) [inline build/Targets/Dependencies.props]: Same for xUnit, we have RTW packages available now.

File: src/NuGet/BuildNuGets.csx [hunk 5]
```diff
@@ -212,8 +213,27 @@ int PackFiles(string[] nuspecFiles, string licenseUrl)
...
+            // HACK: since Microsoft.Net.Compilers 2.0.0 was uploaded by accident and later deleted, we must bump the minor.
+            // We will do this to both the regular Microsoft.Net.Compilers package and also the netcore package to keep them
+            // in sync.
+            if (BuildVersion.StartsWith("2.0.") && Path.GetFileName(file).StartsWith("Microsoft.Net.Compilers."))
+            {
+                string[] buildVersionParts = BuildVersion.Split('-');
+                string[] buildVersionBaseParts = buildVersionParts[0].Split('.');
+                
+                buildVersionBaseParts[buildVersionBaseParts.Length - 1] =
+                    (int.Parse(buildVersionBaseParts[buildVersionBaseParts.Length - 1]) + 1).ToString();
```
Comment (tannergooding) [inline src/NuGet/BuildNuGets.csx]: `TryParse` and fail if that doesn't work?

File: build/Targets/Dependencies.props (hunk 4 above)
Comment (jasonmalinowski) [inline build/Targets/Dependencies.props]: Won't fix at this point. The package change we're making is known to have the same contents otherwise.

File: src/NuGet/BuildNuGets.csx (hunk 5 above)
Comment (jasonmalinowski) [inline src/NuGet/BuildNuGets.csx]: Don't see how I'd fail in any way other than throwing an exception anyways. (No, I'm not logging an error and continuing. Errors should be fatal here.)

**Comment by jasonmalinowski** [comment]
Since Elfie has been published, this should now pass. @dotnet-bot retest this please.

An FYI to @MattGertz we are merging this PR which contains some of the fixes we needed to do to make our NuGet packages. We manually repacked our NuGet packages to account for these, but merging these means any future fixes we need to take will work automatically.

**Comment by MattGertz** [comment]
Understood, thanks for the heads up.  This makes sense.

**Comment by jasonmalinowski** [comment]
@dotnet-bot test vsi please.

**Comment by jasonmalinowski** [comment]
@dotnet-bot test vsi please.

**Comment by jasonmalinowski** [comment]
@tannergooding can you give a signoff on this if it looks good?

**Comment by jasonmalinowski** [comment]
@CyrusNajmabadi as an FYI, the changes in 86aea26cb70788f4ff5518b2e0b3d6b6b381a412 we should revert in dev15.1.x and master. I've written it though so if they are merged it'll be a no-op.

**Comment by jasonmalinowski** [comment]
@CyrusNajmabadi: merging this now.
